javascript/integration-tests/test-utils.ts

```diff
@@ -343,3 +345,38 @@ export const lockDealOrder = async (
         throw new Error('LockDealOrder failed');
     }
 };
+
+export const extractFee = async (
+    resolve: any,
+    reject: any,
+    unsubscribe: any,
+    api: ApiPromise,
+    dispatchError: any,
+    events: EventRecord[],
+    status: any,
+): Promise<void> => {
+    expectNoDispatchError(api, dispatchError);
+
+    if (status.isInBlock) {
+        const balancesWithdraw = events.find(({ event: { method, section } }) => {
+            return section === 'balances' && method === 'Withdraw';
+        });
+
+        expect(balancesWithdraw).toBeTruthy();
+
+        if (balancesWithdraw) {
```

Comment:
it is not necessary from a functional POV however the linter (tsc) complains if I remove the if condition: 

```
integration-tests/test-utils.ts:367:22 - error TS2532: Object is possibly 'undefined'.

367         const fee = (balancesWithdraw.event.data[1] as Balance).toBigInt();
                         ~~~~~~~~~~~~~~~~
```

Reply:
`tsc` treats built in control flow specially, so it knows to narrow the type of `balancesWithdraw` from `Thing | undefined` to `Thing` in the if-body since we know that it can't be undefined. The typescript compiler can't "see into" `toBeTruthy()` and know that all of the following lines are unreachable if `balancesWithdraw === undefined`. So even though _we_ know that it's valid, you need to have the if statement so the compiler can narrow.